tests/components/bbox/test_sensor.py

```diff
@@ -0,0 +1,200 @@
+"""Tests for the Bbox sensor platform."""
+
+from unittest.mock import MagicMock, patch
+
+import pytest
+import requests
+
+from homeassistant.components.bbox.sensor import (
+    PLATFORM_SCHEMA,
+    SENSOR_TYPES,
+    SENSOR_TYPES_UPTIME,
+    BboxData,
+    BboxSensor,
+    BboxUptimeSensor,
+    setup_platform,
+)
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.const import CONF_MONITORED_VARIABLES, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+
+@pytest.fixture
+def mock_bbox_api():
+    """Mock the pybbox API."""
+    with patch("homeassistant.components.bbox.sensor.pybbox") as mock_pybbox:
+        mock_box = MagicMock()
+        mock_pybbox.Bbox.return_value = mock_box
+
+        # Mock get_ip_stats response
+        mock_box.get_ip_stats.return_value = {
+            "rx": {
+                "maxBandwidth": 100000000,  # 100 Mbps
+                "bandwidth": 50000000,  # 50 Mbps
+            },
+            "tx": {
+                "maxBandwidth": 50000000,  # 50 Mbps
+                "bandwidth": 25000000,  # 25 Mbps
+            },
+        }
+
+        # Mock get_bbox_info response
+        mock_box.get_bbox_info.return_value = {
+            "device": {
+                "uptime": 3600,  # 1 hour
+                "numberofboots": 5,
+            }
+        }
+
+        yield mock_box
+
+
+@pytest.fixture
+def sensor_config():
+    """Return sensor configuration."""
+    return {
+        SENSOR_DOMAIN: {
+            CONF_MONITORED_VARIABLES: [
+                "down_max_bandwidth",
+                "up_max_bandwidth",
+                "current_down_bandwidth",
+                "current_up_bandwidth",
+                "number_of_reboots",
+                "uptime",
+            ],
+            CONF_NAME: "Test Bbox",
+        }
+    }
+
+
+async def test_setup_platform_success(
+    hass: HomeAssistant, sensor_config: dict, mock_bbox_api
+) -> None:
+    """Test successful sensor platform setup."""
+    add_entities_mock = MagicMock(spec=AddEntitiesCallback)
+
+    setup_platform(hass, sensor_config[SENSOR_DOMAIN], add_entities_mock)
+
+    # Verify entities were added
+    assert add_entities_mock.called
+    entities = add_entities_mock.call_args[0][0]
+    assert len(entities) == 6  # All 6 sensor types
+
+
+async def test_bbox_data_update_success(mock_bbox_api) -> None:
+    """Test successful BboxData update."""
+    bbox_data = BboxData()
```

Comment:
Can you avoid creating a new instance of `BboxData` object in tests?  Instead, you can simply execute the `setup_platforms` method.